packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -25,23 +26,56 @@ final String _ipv6Loopback = InternetAddress.loopbackIPv6.address;
 
 /// Read the log for a particular device.
 class _FuchsiaLogReader extends DeviceLogReader {
-  _FuchsiaLogReader(this._device);
+  _FuchsiaLogReader(this._device, [this._app]);
 
-  // TODO(jonahwilliams): handle filtering log output from different modules.
-  static final Pattern flutterLogOutput = RegExp(r'\[\d+\.\d+\]\[\d+\]\[\d+\]\[klog\] INFO: \w+\(flutter\): ');
+  static final RegExp _flutterLogOutput = RegExp(r'INFO: \w+\(flutter\): ');
+  static final RegExp _utcDateOutput = RegExp(r'\d+\-\d+\-\d+ \d+:\d+:\d+');
 
   FuchsiaDevice _device;
+  ApplicationPackage _app;
+  RegExp _appPatternOutput;
 
   @override String get name => _device.name;
 
   Stream<String> _logLines;
   @override
   Stream<String> get logLines {
-    _logLines ??= fuchsiaSdk.syslogs()
-      .where((String line) => flutterLogOutput.matchAsPrefix(line) != null);
+    _logLines ??= _processLogs(fuchsiaSdk.syslogs());
     return _logLines;
   }
 
+  Stream<String> _processLogs(Stream<String> lines) async* {
+    // Get the starting time of the log processor to filter logs from before
+    // the process attached.
+    final DateTime startTime = systemClock.now();
+    await for (String line in lines) {
```

Comment:
One you move the regex construction outside this loop (see below), this can become:

```dart
await for (String line in lines.where(matchRegExp.hasMatch)) {
```

... then you can remove the `continue` below